app/src/main/java/app/revanced/tiktok/settingsmenu/ReVancedSettingsFragment.java

```diff
@@ -85,10 +85,117 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
+                    SettingsEnum.setValue(SettingsEnum.TIK_HIDE_LIVE, value);
                     return true;
                 });
             }
+            //Hide Story toggle
+            {
+                SwitchPreference preference = new SwitchPreference(context);
+                feedFilter.addPreference(preference);
+                preference.setKey(SettingsEnum.TIK_HIDE_STORY.path);
+                preference.setDefaultValue(SettingsEnum.TIK_HIDE_STORY.defaultValue);
+                preference.setChecked(SettingsEnum.TIK_HIDE_STORY.getBoolean());
+                preference.setTitle("Hide story");
+                preference.setSummary("Hide story from feed.");
+                preference.setOnPreferenceChangeListener((pref, newValue) -> {
+                    final boolean value = (Boolean) newValue;
+                    SettingsEnum.setValue(SettingsEnum.TIK_HIDE_STORY, value);
+                    return true;
+                });
+            }
+            //Hide Image video toggle
+            {
+                SwitchPreference preference = new SwitchPreference(context);
+                feedFilter.addPreference(preference);
+                preference.setKey(SettingsEnum.TIK_HIDE_IMAGE.path);
+                preference.setDefaultValue(SettingsEnum.TIK_HIDE_IMAGE.defaultValue);
+                preference.setChecked(SettingsEnum.TIK_HIDE_IMAGE.getBoolean());
+                preference.setTitle("Hide image video");
+                preference.setSummary("Hide image video from feed.");
+                preference.setOnPreferenceChangeListener((pref, newValue) -> {
+                    final boolean value = (Boolean) newValue;
+                    SettingsEnum.setValue(SettingsEnum.TIK_HIDE_IMAGE, value);
+                    return true;
+                });
+            }
+            {
+
+            }
+            //Filter base on play count
+            {
+                EditTextPreference preference = new EditTextPreference(context);
+                feedFilter.addPreference(preference);
+                preference.setKey(SettingsEnum.TIK_HIDE_PLAY_COUNT_MIN.path);
+                preference.setDefaultValue(SettingsEnum.TIK_HIDE_PLAY_COUNT_MIN.defaultValue);
+                preference.setText(Long.toString(SettingsEnum.TIK_HIDE_PLAY_COUNT_MIN.getLong()));
+                preference.setTitle("Min view count");
+                preference.setSummary("If app show error, please reduce the value or press retry few times");
+                preference.setOnPreferenceChangeListener((pref, newValue) -> {
+                    try {
+                        final long value = Long.parseLong((String) newValue);
+                        SettingsEnum.setValue(SettingsEnum.TIK_HIDE_PLAY_COUNT_MIN, value);
+                        return true;
+                    } catch (NumberFormatException e) {
+                        return false;
+                    }
+                });
```

Comment:
This is quite a lot of duplicate code and can be reduced cleanly. Create an abstract class with all the necessary fields for a preference. Implement the data class for each setting. Save a list of the data classes, map them to preferences, and add them each using `feedFilter.addPreference(preference);`